### DISCUSSION TITLE
Getting Exception after clone repository in VS 2022

### DISCUSSION THREAD

**Discussion by VishvashKumarPatel** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#2756] [Q&A] [answered]
The current .NET SDK does not support targeting .NET 6.0.  Either target .NET 5.0 or lower, or use a version of the .NET SDK that supports .NET 6.0.

![image](https://user-images.githubusercontent.com/21253513/176668304-a1b71013-f1ad-4f69-967f-80f2696271a2.png)

**Comment by Keboo** [comment, marked as answer]
The error typically occurs when you only have the .NET 5 SDK installed.
If you run the command `dotnet --list-sdks` on the command line, you should see the versions of the SDK that you have installed. If you do not have a 6.0.300 or newer, I recommend downloading the latest 6.0 version of the SDK from: https://dotnet.microsoft.com/download

**Reply by BradenSmith21** [reply]
Okay so I have 3.1, 5, and 6 installed yet I still get the same error 
9>C:\Program Files\dotnet\sdk\5.0.408\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.TargetFrameworkInference.targets(141,5): error NETSDK1045: The current .NET SDK does not support targeting .NET 6.0.  Either target .NET 5.0 or lower, or use a version of the .NET SDK that supports .NET 6.0.
9>Done building project "MahAppsDragablzDemo.csproj" -- FAILED.
========== Rebuild All: 0 succeeded, 10 failed, 0 skipped ==========

**Reply by Keboo** [reply]
Form your output it looks like it is trying to compile using the 5.0 SDK. 

Can you share the output of running `dotnet --list-sdks`? Is it possible that your 6.0 installation is a preview SDK?

For example when I run it my output looks like this:
![image](https://user-images.githubusercontent.com/952248/178311532-407a588c-a186-4412-a1ce-8edcdba801d1.png)
When I build it will use the 6.0.301 because that the is the latest minor version that is not a preview (this is controlled by the [global.json file at the root of the repo](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/blob/master/global.json))

**Reply by BradenSmith21** [reply]
![image](https://user-images.githubusercontent.com/88356359/178350271-7a9e60a1-6a72-41d2-ae76-dfded30dbec2.png)

Here is the image, no previews...
My VS is also updated too, I check that first

**Reply by Keboo** [reply]
Interesting, a couple more thoughts from looking at your first screenshot.

1. It looks like VS is running as an Admin. Is it possible that the SDK is not registered as for the admin user? Do you get the same list of SDKs if you run `dotnet --list-sdks` from the Package Manager Console within VS?
2. I am not sure from the screenshot, but can you confirm which version of VS you are running? On older versions (including VS2019) you have to turn on a preview option (Tools>>Options>>Environment>>Preview Features) to get it to pick up .NET 6. One way to test this would be to try and create a new WPF project targeting .NET 6.

**Reply by BradenSmith21** [reply]
![image](https://user-images.githubusercontent.com/88356359/179316594-9f234570-5650-458f-9d5d-1074a7e7c784.png)
I am so sorry I didnt see this message, I was waiting for it for 4 days but it was hidden under the "See more" tab. Here is a screenshot of the previews features (not sure what option to use). I am currently on VS2019, do you recommend switching to 2022, and will that affect my projects currently in development?

**Reply by Keboo** [reply]
Ah I think that helps make some sense of the issue. I think you have a few different options that can work (in no particular order):

1. Simply install VS2022 for use with this library. VS2019 and VS2022 can be installed side by side (both from the Visual Studio installer) and used separately. This way you can keep using VS2019 for your current projects in development and use VS2022 when working with this repository. In many situations you can simply use VS2022 with your VS2019 solutions. There are a few caveats with this. Some changes (such as updating the sln file or using newer language features) may not be backwards compatible. If there are other developers on the project, I often encourage teams to update together.
2. In your VS2019 instance, turn on the "Use previews of the .NET SDK" options (second from the bottom in your screenshot). This checkbox unfortunately has a poor name. It would be more accurate to say "Use versions of the .NET SDK newer than what is shipped with this version of Visual Studio". VS2019 does not know about the .NET 6 SDK so it won't pick it up unless that option is enabled. If you turn that on, it should pick up that SDK and start using it.

**Reply by BradenSmith21** [reply]
Alright, doing #.2 didnt change anything - it was still failing- but I installed VS22 and opened it right up and it ran fine, This is so much help for my program now! Thank you for the quick and informative replies!!
